src/main/java/cms/gov/madie/measure/config/UpdateQICoreTestCaseStratificationsChangeUnit.java

```diff
@@ -0,0 +1,86 @@
+package cms.gov.madie.measure.config;
+
+import cms.gov.madie.measure.repositories.MeasureRepository;
+import gov.cms.madie.models.common.ModelType;
+import gov.cms.madie.models.measure.Measure;
+import gov.cms.madie.models.measure.Stratification;
+import gov.cms.madie.models.measure.TestCasePopulationValue;
+import io.mongock.api.annotations.ChangeUnit;
+import io.mongock.api.annotations.Execution;
+import io.mongock.api.annotations.RollbackExecution;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+@ChangeUnit(id = "qi_core_testcase_strat_update", order = "1", author = "madie_dev")
+public class UpdateQICoreTestCaseStratificationsChangeUnit {
+  @Setter @Getter private List<Measure> tempMeasures;
+
+  @Execution
+  public void updateQICoreTestCaseStratifications(MeasureRepository measureRepository)
+      throws Exception {
+    log.debug("Entering updateQICoreTestCaseStratifications()");
+    List<Measure> measureList = measureRepository.findAllByModel(ModelType.QI_CORE.getValue());
+    log.info("found QI Core measures: {}", measureList.size());
+
+    measureList.forEach(
+        measure -> {
+          if (!measure.isActive()
+              || CollectionUtils.isEmpty(measure.getGroups())
+              || CollectionUtils.isEmpty(measure.getTestCases())) {
+            return;
```

Comment:
I thought foreach can't be broken with break or return statement?  does it really break?

Reply:
Yes, Ashok it does break out of that entry if you use return. I have tested it locally